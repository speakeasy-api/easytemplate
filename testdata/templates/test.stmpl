{{.Global.Test}}
{{.Local.Test}}
{{templateFile "templates/test2.stmpl" "test2.txt" "from test.stmpl"}}
{{toFloatWithPrecision .Local.Value 3}}
{{templateStringInput "hello" "Hello {{.Local}}" "User"}}
{{templateStringMultiple "templates/testMultiple.stmpl" nil 3}}