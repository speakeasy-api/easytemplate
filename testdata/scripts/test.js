require("scripts/include.js");

_.each(["a", "b", "c"], (v) => {
  console.log(v);
});

console.log("Hello easytemplate!");

let values = [1, 2, 3, 4, 5];

let reduced = _.reduce(values, (sum, value) => add(sum, value), 0);

templateFile("templates/test.stmpl", "test.txt", {
  Test: "from test.js",
  Value: multiply(add(reduced, 2), 2),
});
