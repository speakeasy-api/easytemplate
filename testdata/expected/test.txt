global
from test.js

34.000
Hello User
Hello World it's a wonderful day!
Hello World it's a wonderful day! (Isolated)